File: POO/J/OzCorp/src/br/com/ozcorp/business/Role.java
package br.com.ozcorp.business;
public enum Role {
	Director("Director", 25000, Department.Administrative),
	Secretary("Secretary", 7500, Department.Administrative),
	Manager("Manager", 17500, Department.Administrative),
	Engineer("Engineer", 22500, Department.Technical),
	Analyst("Analyst", 15000, Department.DataProcessingCenter);
	Role(String title, double baseSalary, Department department) {
		this.title = title;
		this.baseSalary = baseSalary;
		this.department = department;
	}
	private String title;
	private double baseSalary;
	private Department department;
	
	public double getBaseSalary() {
		return baseSalary;
	}
	
	@Override
	public String toString() {
		return title;
	}
	public String nameOf() {
		return super.toString();
	}
}

File: POO/CS/OzCorp/OzCorp/Business/Role.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OzCorp.Business
{
    public enum Roles : int
    {
        Director,
        Secretary,
        Manager,
        Engineer,
        Analyst
    }
    public struct Role
    {
        public static Role GetRole(Roles Role)
        {
            return roles[(int)Role];
        }
        private static List<Role> roles = new List<Role>()
        {
            new Role("Director", 25000, Department.GetDepartment(Departments.Administrative)),
            new Role("Secretary", 7500, Department.GetDepartment(Departments.Administrative)),
            new Role("Manager", 17500, Department.GetDepartment(Departments.Administrative)),
            new Role("Engineer", 22500, Department.GetDepartment(Departments.Technical)),
            new Role("Analyst", 15000, Department.GetDepartment(Departments.DataProcessingCenter))
        };
        public Role(String Title, decimal BaseSalary, Department Department)
        {
            this.Title = Title;
            this.BaseSalary = BaseSalary;
            this.Department = Department;
        }
        public String Title { get; set; }
        public decimal BaseSalary { get; set; }
        public Department Department { get; set; }
        public override string ToString()
        {
            return Title;
        }
    }
}
